packages/flutter_tools/test/general.shard/android/gradle_test.dart

```diff
@@ -261,6 +252,96 @@ void main() {
     }, overrides: <Type, Generator>{
       FileSystem: () => MemoryFileSystem(),
     });
+
+    testUsingContext('Finds app bundle in release mode - Gradle 3.5', () {
```

Comment:
You can write helper functions to invoke a series of slightly different tests:
```dart
void genGradle35AppBundleTest(String mode) {
  testUsingContext(...);
}
genGradle35AppBundleTest('debug');
genGradle35AppBundleTest('profile');
genGradle35AppBundleTest('release');
```

Reply:
I reduced the boilerplate quite a bit.